include/deal.II/lac/linear_operator.h

```diff
@@ -169,11 +170,12 @@ template <typename Range, typename Domain> class LinearOperator
   LinearOperator (const LinearOperator<Range, Domain> &) = default;
 
   /**
-   * Templated copy constructor that creates a LinearOperator object from an
-   * object @p op for which the conversion function
+   * Templated copy constructor that creates a LinearOperator object from
+   * an object @p op for which the conversion function
    * <code>linear_operator</code> is defined.
    */
-  template<typename Op>
+  template<typename Op,
+           typename = typename std::enable_if<!std::is_base_of<LinearOperator<Range, Domain>, Op>::value>::type>
   LinearOperator (const Op &op)
   {
     *this = linear_operator<Range, Domain, Op>(op);
```

Comment:
There is a negation in the type trait. I want this templated constructor only be available for ```OP``` not derived from ```LinearOperator```. (Default constructor takes care of a derived class).